### PR TITLE
[ET-2018] Classic Editor > ECP upsell showing when it should not

### PR DESCRIPTION
**Pull request by codingmusician**
### 🎫 Ticket
[ET-2018]

### 🗒️ Description
The ECP upsell (for multiple ticket types) should not be showing when ECP is installed and activated. It should also not be showing when editing a non-event. In both cases, the upsell was showing. This PR adds the logic to fix that.

I also added a check to make sure we're in the admin side when editing/adding a ticket. This will prevent the upsell from showing within the Community Tickets interface on the front-end.

### 🎥 Artifacts 
[UPDATED]
https://www.loom.com/share/f105503368504f5592d391deb3a985ac?sid=62acd840-178c-4d09-b9b1-da979f777466

### ✔️ Checklist
- [x] Changelog entry in the `readme.txt` file.
- [ ] Code is covered by **NEW** `wpunit` or `integration` tests.
- [x] Code is covered by **EXISTING** `wpunit` or `integration` tests.
- [x] Are all the **required** tests passing?
- [x] Automated code review comments are addressed.
- [x] Have you added Artifacts?
- [x] Check the base branch for your PR.
- [x] Add your PR to the project board for the release.

[ET-2018]: https://stellarwp.atlassian.net/browse/ET-2018?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Tickets/Admin/Upsell.php [hunk 1]
```diff
@@ -227,6 +227,21 @@ public function maybe_show_paystack_promo( $settings ) {
 	 * @param Template $template Current instance of the Tribe__Template.
 	 */
 	public function render_ticket_type_upsell_notice( string $file, array $name, Template $template ): void {
+		// If they already have ECP activated or are not within the admin area, then bail.
```
Comment (redscar) [inline src/Tickets/Admin/Upsell.php:230]: We should probably update the `@since`.

**Review by rafsuntaskin — CHANGES_REQUESTED** [review]
This hook should be unhooked from ECP instead of doing Class_exist checks inside this one.

Or we can use the Flexible Tickets base provider as well as that is only registered when ECP is enabled: https://github.com/the-events-calendar/event-tickets/blob/7927ad86111bc17e3c3fb2c83db53f9abbbd516e/src/Tickets/Flexible_Tickets/Base.php#L95

File: src/Tickets/Admin/Upsell.php [hunk 2]
```diff
@@ -221,12 +221,23 @@ public function maybe_show_paystack_promo( $settings ) {
...
 	public function render_ticket_type_upsell_notice( string $file, array $name, Template $template ): void {
+		// Check if post type is an event.
+		if ( ! function_exists( 'tribe_is_event' ) || ! tribe_is_event() ) {
+			return;
+		}
+
+		// If not within the admin area, then bail.
+		if ( ! is_admin() ) {
+			return;
+		}
```
Comment (bordoni) [inline src/Tickets/Admin/Upsell.php:239]: I would use an action like `did_action( 'action_that_only_happens_in_pro' )`, that tends to be a safer bet.

**Review by bordoni — DISMISSED** [review]

**Comment by codingmusician** [comment]
Added this check, which is an action that runs once within ECP upon `plugins_loaded`:
```did_action( 'tribe_events_pro_init_apm_filters' )```

**Review by redscar — APPROVED** [review]

**Review by rafsuntaskin — APPROVED** [review]

**Pull request merged**